pkg/controller/certificaterequests/controller.go

```diff
@@ -96,10 +102,17 @@ func (c *Controller) Register(ctx *controllerpkg.Context) (workqueue.RateLimitin
 
 	// build a list of InformerSynced functions that will be returned by the Register method.
 	// the controller will only begin processing items once all of these informers have synced.
-	mustSync := []cache.InformerSynced{
+
+	// Ensure we also catch all extra informers for this certificate controller instance
+	var extraInformersMustSync []cache.InformerSynced
+	for _, i := range c.extraInformers {
+		extraInformersMustSync = append(extraInformersMustSync, i.HasSynced)
+	}
+
+	mustSync := append([]cache.InformerSynced{
 		certificateRequestInformer.Informer().HasSynced,
 		issuerInformer.Informer().HasSynced,
```

Comment:
unrelated to this PR, but shouldn't we also watch ClusterIssuer resources too?

Reply:
Do we not already?

Reply:
Yep, ignore me - it's done below 🙈